backend/src/interface/persistent_data.rs

```diff
@@ -185,22 +167,26 @@ pub trait CommandDataAccess: Sync + Send {
     /// Removes a downvote from the given image.
     async fn remove_downvote(&self, image_id: Uuid, user_id: Uuid) -> Result<()>;
     /// Adds an image link to the database. The image will be related to the given meal.
-    async fn link_image(
-        &self,
-        meal_id: Uuid,
-        user_id: Uuid,
-        image_hoster_id: String,
-        url: String,
-    ) -> Result<()>;
+    async fn link_image(&self, meal_id: Uuid, user_id: Uuid) -> Result<Uuid>;
+
+    /// Reverts the linking of the given image by deleting the link.
+    /// Useful if an error ocurred with the image itself.
+    async fn revert_link_image(&self, image_id: Uuid) -> Result<()>;
```

Comment:
Is there a usage now? For my understanding, images will be linked after evaluation (image_validation component). Users are not able to remove images by themselves. If it is unused we should mark it as unused (for now)